### PR TITLE
homework3

### PR DESCRIPTION
**Pull request by Dmitry-Zv**

### CODE REVIEW

File: app/src/main/java/by/zharikov/myapplication/homework3/Homework3.kt [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import by.zharikov.myapplication.R
+import kotlinx.android.synthetic.main.activity_homework3.*
+
+class Homework3 : AppCompatActivity() {
```
Comment (stasenkots) [inline app/src/main/java/by/zharikov/myapplication/homework3/Homework3.kt:11]: Too many repeated actions.Think about using collections and loops

File: app/src/main/java/by/zharikov/myapplication/homework3/Homework3.kt [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            textView.text = it.toString()
+        })
+        brestViewModel.counterCabbage.observe(this, Observer {
+            textView2.text = it.toString()
+        })
+        brestViewModel.counterBeet.observe(this, Observer {
+            textView3.text = it.toString()
+        })
+
+        val vitebskViewModel = ViewModelProvider(this).get(VitebskViewModel::class.java)
```
Comment (stasenkots) [inline app/src/main/java/by/zharikov/myapplication/homework3/Homework3.kt]: I thing you don't need class for each country bc your parent class do all what you need

**Review by stasenkots — CHANGES_REQUESTED** [review]

File: app/src/main/java/by/zharikov/myapplication/homework3/RegionViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlin.random.Random
+
+abstract class  RegionViewModel() : ViewModel() {
+    var counterPotatoes = MutableLiveData(0)
+    var counterCabbage = MutableLiveData(0)
+    var counterBeet = MutableLiveData(0)
+    var winnerRegion = MutableLiveData(false)
+    var winner = false
```
Comment (andrewKarachun0304) [inline app/src/main/java/by/zharikov/myapplication/homework3/RegionViewModel.kt]: Winner is redundant variable, you can use winnerRegion in while

File: app/src/main/java/by/zharikov/myapplication/homework3/Homework3.kt (hunk 2 above)
Comment (andrewKarachun0304) [reply app/src/main/java/by/zharikov/myapplication/homework3/Homework3.kt:27]: +

Comment (andrewKarachun0304) [reply app/src/main/java/by/zharikov/myapplication/homework3/Homework3.kt]: I agree

File: app/src/main/java/by/zharikov/myapplication/homework3/BrestViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package by.zharikov.myapplication.homework3
+
+import android.content.Context
+import android.widget.Toast
```
Comment (andrewKarachun0304) [inline app/src/main/java/by/zharikov/myapplication/homework3/BrestViewModel.kt]: Don't use in all ViewModel classes

**Review by limitAlltheir — APPROVED** [review]

File: app/src/main/java/by/zharikov/myapplication/homework3/Homework3.kt (hunk 1 above)
Comment (Dmitry-Zv) [reply app/src/main/java/by/zharikov/myapplication/homework3/Homework3.kt:11]: Updated

**Review by Dmitry-Zv — COMMENTED** [review]
Updated

**Review by stasenkots — APPROVED** [review]
Good.But using assert call (!!) is no good.

File: app/src/main/java/by/zharikov/myapplication/homework3/Homework3.kt [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_homework3)
+        val regionViewModel = ViewModelProvider(this).get(RegionViewModel::class.java)
+
+        button_start_harvesting.setOnClickListener {
+
+            regionViewModel.findWinner()
+            regionViewModel.brestRegion.counterPotatoes.observe(this, Observer {
+                textView.text = it.toString()
+            })
```
Comment (TMSlebedev) [inline app/src/main/java/by/zharikov/myapplication/homework3/Homework3.kt:22]: Better to observe whole region, not a single vegetable in order to reduce code

**Review by TMSlebedev — APPROVED** [review]

**Review by andrewKarachun0304 — APPROVED** [review]

**Pull request merged**